src/cata_utility.h

```diff
@@ -72,6 +74,31 @@ double round_up( double val, unsigned int dp );
  */
 bool isBetween( int test, int down, int up );
 
+/** Sums the two terms, being careful to not trigger overflow or underflow
+ * Correctly handles overflow, underflow, and nonintegral types
+ *
+ * @param a The first addend.
+ * @param b The second addend.
+ * @return the sum of the addends, but truncated to std::numeric_limits<T>::max or min.
+ */
+template <typename T>
+static T sum_no_wrap( T a, T b )
+{
+    static_assert( std::is_integral<T>::value, "sum_no_wrap applied to nonintegral type." );
+
+    // Check for overflow
+    if( ( a > 0 ) && ( b > std::numeric_limits<T>::max() - a ) ) {
```

Comment:
The parenthesis around the relational operators are not needed:
```suggestion
    if( a > 0 && b > std::numeric_limits<T>::max() - a ) {
```